fastai/widgets/image_downloader.py

```diff
@@ -0,0 +1,206 @@
+from ..core import *
+from ..vision.data import *
+from ipywidgets import widgets, Layout, Output, HBox, VBox, Text, BoundedIntText, Button, Dropdown, Box
+from IPython.display import clear_output, display
+from urllib.parse import quote
+from bs4 import BeautifulSoup
+import time
+
+__all__ = ['ImageDownloader', 'download_google_images']
+
+class ImageDownloader():
+    """
+    Displays a widget that allows searching and downloading images from google images search
+    in a Jupyter Notebook or Lab.
+    """
+
+    def __init__(self, path:Union[Path,str]='data'):
+        "Setup path to save images to, init the UI, and render the widgets."
+        self._path = Path(path)
+        self._ui = self._init_ui()
+        self.render()
+
+    def _init_ui(self) -> VBox:
+        """
+        Initialize the widget UI and return the UI,
+        only used to initially create all the components and setup event handlers."
+        """
+        self._search_input = Text(placeholder="What images to search for?")
+        self._count_input = BoundedIntText(placeholder="How many pics?",
+                                        value=10, min=1, max=5000, step=1,
+                                        layout=Layout(width='60px'))
+        self._size_input = Dropdown(options= ['>400*300', '>640*480', '>800*600', '>1024*768', '>2MP', '>4MP', '>6MP', '>8MP', '>10MP'],
```

Comment:
Maybe options need to go on several lines here as it's really wide.
Or maybe have the dict defined later be a private dict defined in this module and use {that_dict}.keys() here to get things a little bit shorter?

Reply:
Moved `_img_sizes` to a private variable in the module, using that in both `_search_url` and the UI options. Tidies things up indeed. 👍